great_expectations/profiler/parameter_builder/parameter.py

```diff
@@ -0,0 +1,18 @@
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import IDDict
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.types import SerializableDictDot
+
+
+@dataclass
+class Parameter(IDDict, SerializableDictDot):
```

Comment:
As discussed, I think it makes sense to start to do away with `DictDot` and `SerializableDictDot`, since we can accomplish the same functionality with `dataclasses`. Maybe recreating `SerializableDictDot` as `SerializableDataclass`.

Reply:
@talagluck Agreed -- a comment to this effect, worded almost identically to yours, is now in this module!